CMakeLists.txt

```diff
@@ -704,6 +712,8 @@ if(UNIX)
   target_link_libraries(mxnet PRIVATE mxnet_static)
   target_link_libraries(mxnet_static PUBLIC ${CMAKE_DL_LIBS})
   set_target_properties(mxnet_static PROPERTIES OUTPUT_NAME mxnet)
+  set_target_properties(mxnet PROPERTIES VERSION "${ver_major}.${ver_minor}.${ver_patch}")
```

Comment:
needs this with the last change?

Reply:
https://cmake.org/cmake/help/latest/prop_tgt/VERSION.html does not say anything about the target property being initialized from the project version. So it should still be needed. But you can try it out locally: Don's set the target property but print it. I expect it would print an empty value.

Instead of using `ver_major` you can use https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION_MAJOR.html#variable:PROJECT_VERSION_MAJOR etc
 for clarity

Reply:
like this?

~~~cmake
  set_target_properties(mxnet PROPERTIES VERSION "${PROJECT_VERSION}")
  set_target_properties(mxnet PROPERTIES SOVERSION "${PROJECT_VERSION_MAJOR}")
~~~

Reply:
You don't need the quotation marks around ${PROJECT_VERSION}

Ref: Github search: https://github.com/search?l=CMake&q=set_target_properties%28+PROPERTIES+VERSION+PROJECT_VERSION%29&type=Code

Reply:
without quotes
<details>

![Screenshot_20200318_145003](https://user-images.githubusercontent.com/462213/76967483-d1a34280-6927-11ea-906c-f0dc921da0d5.png)

</details>

with quotes
<details>

![Screenshot_20200318_145123](https://user-images.githubusercontent.com/462213/76967664-01524a80-6928-11ea-8a4c-d3855e2a4f15.png)

</details>

Reply:
other test, ans this surprise me, i just add
~~~cmake
message(STATUS "Project version: ${PROJECT_VERSION}")
~~~
just before `project()` function, and works

![Screenshot_20200318_151002](https://user-images.githubusercontent.com/462213/76969342-8d657180-692a-11ea-95bc-bb7401e81fae.png)

bus is vanished if put that in the bottom/middle of the CMakeLists

![Screenshot_20200318_151423](https://user-images.githubusercontent.com/462213/76969787-2ac0a580-692b-11ea-8eb2-e1fed938adbc.png)
![Screenshot_20200318_151156](https://user-images.githubusercontent.com/462213/76969551-d1587680-692a-11ea-810b-b8d1fba0947a.png)


cmake is a bit complex for me, IDK where is the problem, so, i think is better keep untounch with `${ver_major}.${ver_minor}.${ver_patch}` and in near future (i i found the problem), fix it

greetings

Reply:
ook, i found the problem

`project()` is call twice in the main CMakeFile

https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L9
https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L89

why this?

Reply:
i think is finally done

tested on local and works as spected

Reply:
Thank you for fixing it